api/v1/metric_routes.go

```diff
@@ -37,12 +36,11 @@ func handleGetMetrics(rw http.ResponseWriter, r *http.Request) {
 		t = engine.ExecutionScheduler.GetState().GetCurrentTestRunDuration()
 	}
 
-	metrics := make([]Metric, 0)
-	for _, m := range engine.Metrics {
-		metrics = append(metrics, NewMetric(m, t))
-	}
+	engine.MetricsLock.Lock()
+	metrics := newMetricsJSONAPI(engine.Metrics, t)
+	engine.MetricsLock.Unlock()
```

Comment:
we had a race ?

Reply:
Yes, in the case that the API is hit during the test